### PR TITLE
[TC-1666] - Ten Latest Sboms with vulnerabilities summary backend

### PR DESCRIPTION
**Pull request by m-brophy**
https://issues.redhat.com/browse/TC-1666

### CODE REVIEW

**Review by carlosthe19916 — APPROVED** [review]
@m-brophy my apologies for the late review.

This is the exact JSON I get from the backend:

```json
[
  [
    "sbom1", // array[0] is a string
    [// array[1] is another array
      {
        "severity": "none",
        "count": 3
      },
      {
        "severity": "low",
        "count": 5
      },
      {
        "severity": "medium",
        "count": 10
      },
      {
        "severity": "high",
        "count": 4
      },
      {
        "severity": "critical",
        "count": 2
      }
    ]
  ],
  [
    "sbom2",
    [
      {
        "severity": "none",
        "count": 3
      },
      {
        "severity": "low",
        "count": 5
      },
      {
        "severity": "medium",
        "count": 10
      },
      {
        "severity": "high",
        "count": 4
      },
      {
        "severity": "critical",
        "count": 2
      }
    ]
  ],
  [
    "sbom3",
    [
      {
        "severity": "none",
        "count": 3
      },
      {
        "severity": "low",
        "count": 5
      },
      {
        "severity": "medium",
        "count": 10
      },
      {
        "severity": "high",
        "count": 4
      },
      {
        "severity": "critical",
        "count": 2
      }
    ]
  ]
]
```

If you notice the second array contains "sbom1" and another array in the same array. Could we change it?

I could propose something like:

```json
[
    {
        "sbom_id": "something",
        "sbom_name": "Quarkus-v1",
        "vulnerabilities": {
            "none": 3,
            "low": 3,
            "medium": 0,
            "high": 3,
            "critical": 3
        }
    }
]
```

**Review by carlosthe19916 — CHANGES_REQUESTED** [review]
.

**Comment by carlosthe19916** [comment]
@m-brophy I have created a PR to your branch with what I think is  working version of "last 10 sboms with vulnerabilities". If it helps please take a look at it, there might be place for improvement and of course we should test and check if it works or not https://github.com/m-brophy/trustification/pull/1

File: bombastic/api/src/server.rs [hunk 1]
```diff
@@ -408,3 +409,61 @@ async fn delete_sboms(
...
+    (status = 200, description = "Search completed"),
+    (status = BAD_REQUEST, description = "Bad query"),
+    (status = 401, description = "Not authenticated"),
+    ),
+    params(
+    ("q" = String, Query, description = "Search query"),
+    )
+)]
+#[get("/sbom/status")]
+async fn sbom_status(
```
Comment (bxf12315) [inline bombastic/api/src/server.rs:430]: I have a suggestion: there is no need to create a new endpoint, we can just utilize the existing one.[1] 

[1] https://github.com/trustification/trustification/blob/621fc8abc1b67a25686e635a1f9ac5319a845b82/bombastic/api/src/server.rs#L235

**Review by carlosthe19916 — APPROVED** [review]

**Comment by carlosthe19916** [comment]
Thanks for the PR https://github.com/trustification/trustification/pull/1716 @mrizzi !

**Pull request merged**